scripts/tasks/compile.ts

```diff
@@ -0,0 +1,39 @@
+import { readFile } from 'fs-extra';
+import { resolve } from 'path';
+
+import { maxConcurrentTasks } from '../utils/maxConcurrentTasks';
+import { exec } from '../utils/exec';
+import type { Task } from '../task';
+
+const linkedContents = `export * from '../src/index'`;
+const linkCommand = `nx run-many --target="prep" --all --parallel --exclude=@storybook/addon-storyshots,@storybook/addon-storyshots-puppeteer -- --reset`;
+const noLinkCommand = `nx run-many --target="prep" --all --parallel=8 ${
+  process.env.CI ? `--max-parallel=${maxConcurrentTasks}` : ''
+} -- --reset --optimized`;
+
+export const compile: Task = {
+  description: 'Compile the source code of the monorepo',
+  dependsOn: ['install'],
+  async ready({ codeDir }, { link }) {
+    try {
+      const contents = await readFile(resolve(codeDir, './lib/store/dist/index.d.ts'), 'utf8');
+      const isLinkedContents = contents.indexOf(linkedContents) !== -1;
+      if (link) return isLinkedContents;
+      return !isLinkedContents;
```

Comment:
I think having a comment about why we pick this one random library, would be good for future me/us.